godot-core/src/obj/base.rs

```diff
@@ -36,13 +39,19 @@ pub struct Base<T: GodotClass> {
 impl<T: GodotClass> Base<T> {
     /// # Safety
     /// The returned Base is a weak pointer, so holding it will not keep the object alive. It must not be accessed after the object is destroyed.
+    ///
+    /// While not necessary for safety, `base` is expected to be alive to prevent logic errors, i.e. user `init()` must not have run yet.
```

Comment:
as this is written in the `# Safety` documentation it's a bit unclear if this means:
1. `base` must be alive when this method is called (this i believe must be true since `Gd::from_obj_sys_weak` requires that)
2. `base` must be alive while the returned `Base<T>` object exists (this is false, as it's generally safe for this to be violated it's just a logic bug)